### PR TITLE
Get all flashes

### PR DESCRIPTION
**Pull request by geerteltink**
- [x] Are you creating a new feature?
  - [x] Why is the new feature needed? What purpose does it serve?
  - [x] Base your feature on the `develop` branch, and submit against that branch.
  - [x] Add only one feature per pull request; split multiple features over multiple pull requests
  - [x] Add tests for the new feature.
  - [x] Add documentation for the new feature.

Sometimes you don't know which messages are added. `FlashMessages::getFlashes()` retrieves an array with all messages for the current request.

### CODE REVIEW

File: docs/book/messages.md [hunk 1]
```diff
@@ -61,6 +61,19 @@ interface FlashMessagesInterface
...
+    /**
+     * Retrieve all flash values.
+     *
+     * Will return all values was set in a previous request, or if `flashNow()`
+     * was called in this request with the same `$key`.
+     *
+     * WILL NOT return values set in the current request via `flash()`.
+     *
+     * @param mixed $default Default value to return if no flash value exists.
+     * @return mixed
```
Comment (michalbundyra) [inline docs/book/messages.md]: `mixed` or `array` ? or `mixed[]`?

File: docs/book/messages.md [hunk 2]
```diff
@@ -61,6 +61,19 @@ interface FlashMessagesInterface
...
 
+    /**
+     * Retrieve all flash values.
+     *
+     * Will return all values was set in a previous request, or if `flashNow()`
+     * was called in this request with the same `$key`.
+     *
+     * WILL NOT return values set in the current request via `flash()`.
+     *
+     * @param mixed $default Default value to return if no flash value exists.
```
Comment (michalbundyra) [inline docs/book/messages.md]: Can't see any param defined in the function signature.

File: docs/book/messages.md [hunk 3]
```diff
@@ -61,6 +61,19 @@ interface FlashMessagesInterface
      */
     public function getFlash(string $key, $default = null);
 
+    /**
+     * Retrieve all flash values.
```
Comment (michalbundyra) [inline docs/book/messages.md:65]: I'm not a big fan of copying code into the documentation, then we need to maintain it in two places, and always remember to update it in the docs.... I know it was already here and you've just updated it 👍

File: docs/book/messages.md [hunk 4]
```diff
@@ -61,6 +61,19 @@ interface FlashMessagesInterface
...
+     *
+     * Will return all values was set in a previous request, or if `flashNow()`
+     * was called in this request with the same `$key`.
```
Comment (michalbundyra) [inline docs/book/messages.md]: It is not clear what is the `$key`.

File: src/FlashMessages.php [hunk 5]
```diff
@@ -119,6 +119,22 @@ public function getFlash(string $key, $default = null)
...
+    /**
+     * Retrieve all flash values.
+     *
+     * Will return all values was set in a previous request, or if `flashNow()`
+     * was called in this request with the same `$key`.
+     *
+     * WILL NOT return values set in the current request via `flash()`.
+     *
+     * @param mixed $default Default value to return if no flash value exists.
+     * @return mixed
```
Comment (michalbundyra) [inline src/FlashMessages.php]: The same comments here as in the docs

File: src/FlashMessagesInterface.php [hunk 6]
```diff
@@ -64,6 +64,19 @@ public function flashNow(string $key, $value, int $hops = 1) : void;
...
+    /**
+     * Retrieve all flash values.
+     *
+     * Will return all values was set in a previous request, or if `flashNow()`
+     * was called in this request with the same `$key`.
+     *
+     * WILL NOT return values set in the current request via `flash()`.
+     *
+     * @param mixed $default Default value to return if no flash value exists.
+     * @return mixed
```
Comment (michalbundyra) [inline src/FlashMessagesInterface.php]: And here as well.

File: test/TestAsset/FlashMessages.php [hunk 7]
```diff
@@ -42,6 +42,10 @@ public function getFlash(string $key, $default = null)
     {
     }
 
+    public function getFlashes() : array
+    {
+    }
```
Comment (michalbundyra) [inline test/TestAsset/FlashMessages.php]: This method definition is wrong, must return `array`, but it's `void`. Not sure where (if) it is used?

**Review by michalbundyra — CHANGES_REQUESTED** [review]

**Review by michalbundyra — APPROVED** [review]

**Comment by vitorloureiro** [comment]
Hi @xtreamwayz 
It is possible to merge this? i think this is essential for the flash messages middleware.

**Comment by geerteltink** [comment]
Yes, it's possible after a review from @weierophinney :)

File: src/FlashMessages.php [hunk 8]
```diff
@@ -119,6 +119,21 @@ public function getFlash(string $key, $default = null)
...
 
+    /**
+     * Retrieve all flash values.
+     *
+     * Will return all values was set in a previous request, or if `flashNow()`
+     * was called in this request.
+     *
+     * WILL NOT return values set in the current request via `flash()`.
+     *
+     * @return array
```
Comment (michalbundyra) [inline src/FlashMessages.php:130]: Can be removed, as we have it already in method signature.

File: src/FlashMessagesInterface.php [hunk 9]
```diff
@@ -64,6 +64,18 @@ public function flashNow(string $key, $value, int $hops = 1) : void;
...
 
+    /**
+     * Retrieve all flash values.
+     *
+     * Will return all values was set in a previous request, or if `flashNow()`
+     * was called in this request.
+     *
+     * WILL NOT return values set in the current request via `flash()`.
+     *
+     * @return array
```
Comment (michalbundyra) [inline src/FlashMessagesInterface.php:75]: The same as above, can be removed.

**Pull request merged**